src/lib/users/ssh_public_key.rb

```diff
@@ -0,0 +1,76 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast2/execute"
+
+module Y2Users
+  # This class is a simplified representation of a OpenSSH public key.
+  #
+  # @example Read a public key
+  #   key = Y2Users::SSHPublicKey.new(File.read("id_rsa.pub"))
+  #   key.fingerprint # => "SHA256:uadPyDQj9VlFZVjK8UNp57jOnWwzGgKQJpeJEhZyV0I"
+  class SSHPublicKey
```

Comment:
Again, would it make sense to share this class??

Reply:
I would keep it locally and move it when needed (as the `SSHAuthorizedKeys` module).